code/game/objects/items/devices/slide_projector.dm

```diff
@@ -0,0 +1,204 @@
+/obj/item/storage/slide_projector
+	name = "slide projector"
+	desc = "A handy device capable of showing an enlarged projection of whatever you can fit inside."
+	desc_info = "You can use this in hand to open the interface, click-dragging it to you also works. Click anywhere with it in your hand to project at that location. Click dragging it to that location also works."
+	icon = 'icons/obj/projector.dmi'
+	icon_state = "projector0"
+	max_w_class = ITEMSIZE_SMALL
+	max_storage_space = 10
+	use_sound = 'sound/items/storage/toolbox.ogg'
+	var/static/list/projection_types = list(
+		/obj/item/photo = /obj/effect/projection/photo,
+		/obj/item/paper = /obj/effect/projection/paper,
+		/obj/item = /obj/effect/projection
+	)
+	var/obj/item/current_slide
+	var/obj/effect/projection/projection
+
+/obj/item/storage/slide_projector/Destroy()
+	set_slide(null)
+	return ..()
+
+/obj/item/storage/slide_projector/update_icon()
+	icon_state = "projector[!!projection]"
+
+/obj/item/storage/slide_projector/remove_from_storage(obj/item/W)
+	. = ..()
+	if(. && W == current_slide)
+		set_slide(length(contents) ? contents[1] : null)
+
+/obj/item/storage/slide_projector/handle_item_insertion(var/obj/item/W)
+	. = ..()
+	if(. && !current_slide)
+		set_slide(W)
+
+/obj/item/storage/slide_projector/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	if(!current_slide)
+		to_chat(user, SPAN_WARNING("\The [src] does not have a slide loaded."))
+		return
+	project_at(get_turf(target))
+
+/obj/item/storage/slide_projector/MouseDrop(atom/over)
+	if(use_check_and_message(usr))
+		return
+
+	if(over == usr)
+		interact(usr)
+		return
+
+	var/turf/T = get_turf(over)
+	if(istype(T))
+		afterattack(over, usr)
+
+/obj/item/storage/slide_projector/proc/set_slide(obj/item/new_slide)
+	current_slide = new_slide
+	playsound(loc, 'sound/machines/slide_change.ogg', 50)
+	if(projection)
+		project_at(get_turf(projection))
```

Comment:
Check the original Baystation PR as it has some good concerns regarding this, e.g setting a slide to null not actually removing or deleting it.